extensions/servlet/test/com/google/inject/servlet/ServletDefinitionPathsTest.java

```diff
@@ -216,35 +229,39 @@ protected void service(HttpServletRequest servletRequest,
 
   // Data-driven test.
   public final void testPathInfoWithRegexMatching() throws IOException, ServletException {
+
     // first a mapping of /*
-    pathInfoWithRegexMatching("/path/index.html", "/path", "/(.)*", "/index.html", "");
-    pathInfoWithRegexMatching("/path//hulaboo///index.html", "/path", "/(.)*",
+    pathInfoWithRegexMatching("/path/index.html", "/path", "(/).*", "/index.html", "");
+    pathInfoWithRegexMatching("/path//hulaboo///index.html", "/path", "(/).*",
         "/hulaboo/index.html", "");
-    pathInfoWithRegexMatching("/path/", "/path", "/(.)*", "/", "");
-    pathInfoWithRegexMatching("/path////////", "/path", "/(.)*", "/", "");
+    pathInfoWithRegexMatching("/path/", "/path", "(/).*", "/", "");
+    pathInfoWithRegexMatching("/path////////", "/path", "(/).*", "/", "");
 
     // a servlet mapping of /thing/*
     pathInfoWithRegexMatching("/path/thing////////", "/path", "/thing/(.)*", "/", "/thing");
-    pathInfoWithRegexMatching("/path/thing/stuff", "/path", "/thing/(.)*", "/stuff", "/thing");
-    pathInfoWithRegexMatching("/path/thing/stuff.html", "/path", "/thing/(.)*", "/stuff.html",
+    pathInfoWithRegexMatching("/path/thing/stuff", "/path", "/thing(/).*", "/stuff", "/thing");
+    pathInfoWithRegexMatching("/path/thing/stuff.html", "/path", "/thing(/).*", "/stuff.html",
         "/thing");
-    pathInfoWithRegexMatching("/path/thing", "/path", "/thing/(.)*", null, "/thing");
+    pathInfoWithRegexMatching("/path/thing", "/path", "/thing(/).*", null, "/thing");
 
     // *.xx style mapping
-    pathInfoWithRegexMatching("/path/thing.thing", "/path", ".*\\.thing", null, "/thing.thing");
-    pathInfoWithRegexMatching("/path///h.thing", "/path", ".*\\.thing", null, "/h.thing");
-    pathInfoWithRegexMatching("/path///...//h.thing", "/path", ".*\\.thing", null,
+    //FIXME: what sense make the regex here anyway?
+    pathInfoWithRegexMatching("/path/thing.thing", "/path", "", null, "/thing.thing");
+    pathInfoWithRegexMatching("/path///h.thing", "/path", "", null, "/h.thing");
+    pathInfoWithRegexMatching("/path///...//h.thing", "/path", "", null,
         "/.../h.thing");
-    pathInfoWithRegexMatching("/path/my/h.thing", "/path", ".*\\.thing", null, "/my/h.thing");
+    pathInfoWithRegexMatching("/path/my/h.thing", "/path", "", null, "/my/h.thing");
```

Comment:
because a regular expression without group in the regular expression matcher makes no sense at all. The regular expression matcher doesn't work very intuitive. See the logic in RegexUriPatternMatcher.